apps/mochi-web/pages/profile/transactions/index.tsx

```diff
@@ -52,14 +113,143 @@ const AppPageHeader = () => {
   )
 }
 
+const MAX_TRANSACTION_TO_DISPLAY = 30
+
 const App: NextPageWithLayout = () => {
   const { profile } = useLoginWidget()
+  const { query } = useRouter()
+  // const [ ] = useState()
+  const { transactions: _transactions, isLoading } = useFetchProfileTransaction(
+    profile?.id ?? '',
+    Boolean(profile?.id),
+  )
+  const transactions = useMemo(
+    () => _transactions?.slice(0, MAX_TRANSACTION_TO_DISPLAY),
+    [_transactions],
+  )
+  const [filterType, filterPlatform] = useMemo(() => {
+    let validType = 'all'
+    let validPlatform = 'all'
+
+    if (typeof query.type === 'string') {
+      validType = isValidFilterType(
+        typeof query.type === 'string' ? query.type : '',
+      )
+        ? query.type
+        : validType
+    }
+    if (typeof query.platform === 'string') {
+      validPlatform = isValidFilterPlatform(query.platform)
+        ? query.platform
+        : validPlatform
+    }
+    return [validType, validPlatform] as [
+      TransactionTypeFilterKey,
+      TransactionPlatformFilterKey,
+    ]
+  }, [query.platform, query.type])
+
+  const filteredTransactions = useMemo(() => {
+    return transactions
+      ?.filter((t) => actionTypeFilter(t.action ?? '', filterType))
+      .filter((t) => platformFilter(t.source_platform ?? '', filterPlatform))
+  }, [filterPlatform, filterType, transactions])
 
   return (
     <>
       <SEO title={`${profile?.profile_name}'s transactions`} />
-      <AuthLayout pageHeader={<AppPageHeader />}>
-        <Table columns={[]} data={[]} />
+      <AuthLayout
+        pageHeader={
+          <AppPageHeader
+            filterType={filterType}
+            filterPlatform={filterPlatform}
+          />
+        }
+      >
+        <Table<ModelProfileTransaction>
```

Comment:
Can we add an empty view for the table?

Reply:
@zlatanpham Should we add an empty UI to the table? 
![image](https://github.com/consolelabs/mochi-ui/assets/44285614/0f882943-097f-4cfa-a516-8be30443f38b)